buildSrc/src/main/java/org/elasticsearch/gradle/VersionProperties.java

```diff
@@ -27,11 +27,25 @@ public static String getLucene() {
     private static final Map<String, String> versions = new HashMap<String, String>();
     static {
         Properties props = getVersionProperties();
-        elasticsearch = Version.fromString(props.getProperty("elasticsearch"));
+        Version baseVersion = Version.fromString(props.getProperty("elasticsearch"));
+        if (baseVersion.isSnapshot()) {
+            throw new IllegalArgumentException("version.properties can't contain a snapshot version for elasticsearch");
+        }
+        if (baseVersion.getSuffix().isEmpty() == false) {
+            throw new IllegalArgumentException("version.properties can't contain a version suffix for elasticsearch");
+        }
+        elasticsearch = new Version(
+            baseVersion.getMajor(),
+            baseVersion.getMinor(),
+            baseVersion.getRevision(),
+            System.getProperty("build.version_qualifier", "alpha1"),
```

Comment:
This should not default to `alpha1`. It should default to the empty string. In the production of our nightly snapshots we want to set this externally now and we will start with `alpha1`.

Reply:
I think that this also means that we need to percolate this through to CI otherwise the builds will be looking for ML 7.0.0-SNAPSHOT artifacts which will not exist.

Reply:
One reason that I do not like the default of `alpha1` and instead prefer this to be completely managed externally is that we can end up with the following sequence:
 - today we are producing `alpha1` snapshots
 - at some point we will bump to producing `alpha2` snapshots
 - with this default, we would still be running `alpha1` in CI and that's confusing and wrong (wrong because we need it to be `alpha2` to pick up the latest ML builds)

Reply:
I think the confusion is that without any qualifier, we'll be producing `7.0.0-SNAPSHOT` which is not something we produce now AFAIK, as we go `7.0.0-alpha1-SNAPSHOT` ->  e.x `7.0.0-rc3-SNAPSHOT` -> `7.0.0-rc3` -> `7.0.0`. In this context `7.0.0-SNAPSHOT`  is a pre-alpha1 .
We need to check the `Version` class in the build for this, since I don't think it compares like that.
It will a bit awkward  to start with `7.0.0-SNAPSHOT` end with `7.0.0.` and have all the qualifiers in between, but I think it's the only way to have a default to no qualifier. That means CI will use this default while we are in development mode, and start using the `alpha1` qualifier close to the release only. We don't want to force people building outside of CI to pass this, and knowing this always passing it in CI feels like to big of a difference to me. If we do this, a good way to percolate this to CI is to merge it like this, with `alpha1` as default, then start changing it to empty in CI and eventually change the code to default to empty and remove the parameter from CI.

An alternative is to continue having the qualifier in `version.properties`, but also allow for a property. This way the bump from `-alpha1` to `alpha2` all the way to `rc3` will be a code change,
but if we are happy with `rc3` our job to produce the final release will _not_ do a code change but pass in the qualifier using the property to override the one in `version.properties`.  I think this might be a more straight forward approach as it avoids `7.0.0-SNAPSHOT`. We can even enforce that an empty qualifier can't be a snapshot. It's also easier to propagate and less disruptive overall.

Reply:
I am starting to come around to the view that we simply always produce unqualified builds, yet from time to time we will ship a qualified build with a progression of `alpha1`, `alpha2`, etc. as we head towards GA. This feels like the simplest approach and I am not seeing any major downsides? We would still need to bake the qualifier into the version ID when a qualified build is produced though.

Reply:
@jasontedor you  mean to have CI always build a snapshot without a qualifier like `7.0.0-SNAPSHOT`  and then when we want to ship something we would have one with a qualifier only then e.x `7.0.0-alpha1` so we'll have snapshots before and after the qualified build without distinguishing them. Snapshot and qualifier will become mutually exclusive.  

I like it how that simplifies things, and I think it is ok since we either don't consider the qualifiers already ( e.x. we remove qualified versions from bwc tests ) or we are making changes to compare without the version qualifier ( e.x. when comparing build with `Version.java` and when comparing plugin vs es version ). The only risk I can think of is that we will never run those qualified builds that we make time to time trough our testing. _Is there already some sanity tests that runs on artifacts produced by the RM ?_ That being said, this isn't entirely new,  it's just one way there could be differences between elasticsearch ci and RM and the version comparison code is easy to unit test to make sure it does the right thing.